R/statistic-histogram.R

```diff
@@ -73,59 +99,70 @@ histogram.getParameters <- function(n.bins, n, accuracy, stability, delta=10^-6,
 	return(eps)
 }
 
-#' Histogram confidence interval
+#' Histogram confidence intervals
 #' 
-#' Return the confidence interval for the differentially private histogram release given the
-#' accuracy.
+#' Return the confidence interval for each bins of the differentially private histogram
+#' release, given the accuracy.
+#' 
+#' A confidence interval indicates the range in which we estimate the true value of a
+#' statistic to be. In this case, the confidence interval indicates the range in which
+#' the true count for each histogram bin could be. To give an example: say a 
+#' differentially private release of a histogram bucket has a count of 5. Say the 
+#' confidence interval for that bucket is [3,7]. We can say "we are 95% confident that 
+#' the true count of this histogram bin is between 3 and 7."
 #'
 #' @param release A numeric vector with a noisy estimate of bin counts.
 #' @param n.bins A numeric vector of length one specifying the number of cells 
 #'    in which to tabulate values.
 #' @param n A numeric vector of length one specifying the number of
 #'    observations in in the data.
-#' @param accuracy A numeric vector representing the accuracy needed to 
-#'    guarantee (percent).
+#' @param accuracy A numeric vector representing the accuracy guarantee.
 #'    
 #' @return Confidence interval for the noisy counts in each bin.
 #' @rdname histogram.getCI
 
 histogram.getCI <- function(release, n.bins, n, accuracy) {
-    release <- as.numeric(release)
-    accxn <- accuracy * n
+    release_asNumeric <- as.numeric(release)
     out <- list()
     for (k in 1:n.bins) {
-        bin.count <- release[k]
+        bin.count <- release_asNumeric[k]
         if (bin.count == 0) {
-            out[[k]] <- c(0, accxn)
+            out[[k]] <- c(0, accuracy)
         } else {
-            out[[k]] <- c(max(0, bin.count - accxn), accxn + bin.count)
+            out[[k]] <- c(max(0, bin.count - accuracy), accuracy + bin.count)
         }
     }
     out <- data.frame(do.call(rbind, out))
-    #names(out) <- c('lower', 'upper')
+    names(out) <- c('lower bound', 'upper bound')
     rownames(out) <- names(release)
     return(out)
 }
 
 
-#' Format the release of private histogram
-#'
-#' Convert the release from a table to a data frame
+#' Convert normalized histogram release to data frame
+#' 
+#' Convert the release of the private histogram from a table to a data frame.
+#' While converting, the normalizeHistogram method is called, which normalizes
+#' the private histogram so that the sum of the bin counts is equal to n.
 #'
 #' @param release Table, the result of \code{fun.hist}
 #' @param n Sample size
 #' @return Data frame
 
-histogram.formatRelease <- function(release, n) {
+normalizeReleaseAndConvertToDataFrame <- function(release, n) {
```

Comment:
This is great, but we all agree that this should write to post process field, and not replace raw release.

Reply:
This has been opened as its own issue: https://github.com/privacytoolsproject/PSI-Library/issues/83
And it will be resolved in a separate pull request